src/core/cuda_common_cuda.cu

```diff
@@ -98,145 +83,25 @@ void _cuda_check_errors(const dim3 &block, const dim3 &grid,
   }
 }
 
-__device__ unsigned int getThreadIndex() {
-
-  return blockIdx.y * gridDim.x * blockDim.x + blockDim.x * blockIdx.x +
-         threadIdx.x;
-}
-
-/** Kernel for the initialisation of the particle force array
- * @param[out] particle_forces_device    Local particle force
- * @param[out] particle_torques_device   Local particle torque
- */
-__global__ void init_particle_force(float *particle_forces_device,
-                                    float *particle_torques_device) {
-
-  unsigned int part_index = getThreadIndex();
-
-  if (part_index < global_part_vars_device->number_of_particles) {
-    particle_forces_device[3 * part_index + 0] = 0.0f;
-    particle_forces_device[3 * part_index + 1] = 0.0f;
-    particle_forces_device[3 * part_index + 2] = 0.0f;
-
-#ifdef ROTATION
-    particle_torques_device[3 * part_index] = 0.0f;
-    particle_torques_device[3 * part_index + 1] = 0.0f;
-    particle_torques_device[3 * part_index + 2] = 0.0f;
-#endif
-  }
-}
-
-/** Kernel for the initialisation of the particle force array
- * @param[out] particle_forces_device    Local particle force
- * @param[out] particle_torques_device   Local particle torque
- */
-__global__ void reset_particle_force(float *particle_forces_device,
-                                     float *particle_torques_device) {
-
-  unsigned int part_index = getThreadIndex();
-
-  if (part_index < global_part_vars_device->number_of_particles) {
-    particle_forces_device[3 * part_index + 0] = 0.0f;
-    particle_forces_device[3 * part_index + 1] = 0.0f;
-    particle_forces_device[3 * part_index + 2] = 0.0f;
-#ifdef ROTATION
-    particle_torques_device[3 * part_index + 0] = 0.0f;
-    particle_torques_device[3 * part_index + 1] = 0.0f;
-    particle_torques_device[3 * part_index + 2] = 0.0f;
-#endif
+template <class T> void resize_or_replace(device_vector<T> &vec, size_t n) {
```

Comment:
please add a docstring

Reply:
The docstring would be very similar to the function signature, are you sure
it is required for an internal function?

Reply:
I had the need of existence for this function in mind which should be documented as you have done by now.